sections/predictive-search.liquid

```diff
@@ -1,71 +1,149 @@
+                <a href="{{ article.url }}" class="predictive-search__item link link--text" tabindex="-1">
+                  <div class="predictive-search__item-content predictive-search__item-content--centered">
+                    <h3 class="predictive-search__item-heading h5">{{ article.title }}</h3>
+                  </div>
+                </a>
+              </li>
+            {%- endfor -%}
+          </ul>
+        </div>
+      {%- endif -%}
+      {%- if first_column_results_size > 0 -%}
+        </div>
+      {%- endif -%}
+      {%- if predictive_search.resources.products.size > 0 or predictive_search.resources.articles.size > 0 or predictive_search.resources.pages.size > 0 -%}
+        <div class="predictive-search__result-group">
+          {%- if predictive_search.resources.products.size > 0 -%}
+            <div>
+              <h2 id="predictive-search-products" class="predictive-search__heading text-body caption-with-letter-spacing">
+                {{ 'templates.search.products' | t }}
+                <svg aria-hidden="true" focusable="false" role="presentation" class="spinner" viewBox="0 0 66 66" xmlns="http://www.w3.org/2000/svg">
+                  <circle class="path" fill="none" stroke-width="6" cx="33" cy="33" r="30"></circle>
+                </svg>
+              </h2>
+              <ul id="predictive-search-results-products-list" class="predictive-search__results-list list-unstyled" role="group" aria-labelledby="predictive-search-products">
+                {%- for product in predictive_search.resources.products -%}
+                  <li id="predictive-search-option-product-{{ forloop.index }}" class="predictive-search__list-item" role="option" aria-selected="false">
+                    <a href="{{ product.url }}" class="predictive-search__item predictive-search__item--link-with-thumbnail link link--text" tabindex="-1">
+                      {%- if product.featured_media != blank -%}
+                        <img class="predictive-search__image" src="{{ product.featured_media | image_url: width: 150 }}" alt="{{ product.featured_media.alt }}" width="50" height="{{ 50 | divided_by: product.featured_media.preview_image.aspect_ratio }}">
+                      {%- endif -%}
+                      <div class="predictive-search__item-content{% unless settings.predictive_search_show_vendor or settings.predictive_search_show_price %} predictive-search__item-content--centered{% endunless %}">
+                        {%- if settings.predictive_search_show_vendor -%}
+                          <span class="visually-hidden">{{ 'accessibility.vendor' | t }}</span>
+                          <div class="predictive-search__item-vendor caption-with-letter-spacing">{{ product.vendor }}</div>
+                        {%- endif -%}
+                        <h3 class="predictive-search__item-heading h5">{{ product.title }}
+                        </h3>
```

Comment:
```suggestion
                        <p class="predictive-search__item-heading h5">{{ product.title }}</p>
```

Reply:
I wouldn't say I disagree with this suggestion, but just FYI it looks like during a11y review of the original implementation of predictive search it was specifically [recommended](https://github.com/Shopify/dawn/pull/514#discussion_r698783581) that these product titles use h3. Though I could maybe see that being more relevant if the vendor or price is also showing. If we're confident about this change though, feel free.

Reply:
> do you recall why you recommended an h3 in this past commit?

I do not. As mentioned, the heading structure is not conveyed here. Swap for a generic container.